### ISSUE TITLE
The branch parametric does not build under ghc 8.6

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#3704]
The branch parametric does not build under ghc 8.6.
It would be good to keep agda-parametric alive; we have submitted a paper that relies on it. :)

**Comment by asr** [comment]
Note that the `parametric` branch doesn't build with various versions of GHC (8.2.2, 8.4.4 and 8.6.4) currently supported. It seems it would take less time if someone *familiarised* with this  branch merge the current `master` and fix the conflicts.

**Comment by Saizan** [comment]
I don't know if a straight up merge is an option, things have diverged considerably. Fortunately the changes weren't that huge, so there should be a refactor and integration effort.

In the short term it would be great if there was a good way to identify ghc-compatibility commits to cherry-pick.

**Comment by asr** [comment]
> In the short term it would be great if there was a good way to identify ghc-compatibility commits to cherry-pick.

I don't think so. It seems the branch was tested with GHC 8.0.* and we have added to many changes for supporting the new versions of GHC.

**Comment by asr** [comment]
I guess it is more easy incrementally to add/remove recent/old versions of GHC. *Some* commits necessaries for this job are in the following issues:

Remove support for GHC 7.6.3: #1972
Remove support for GHC 7.8.4: #2692
Support GHC 8.0.2: #2319
Support GHC 8.2.1: #2540
Support GHC 8.4.1: #2878
Support GHC 8.4.2: #3023
Support GHC 8.4.3: #3096
Support GHC 8.4.4: #3273
Support GHC 8.6.1: #3160
Support GHC 8.6.2: #3368
Support GHC 8.6.3: #3442
Support GHC 8.6.4: #3610
Support GHC 8.6.5: #3685

**Comment by L-TChen** [comment]
Is there still a plan to merge the branch `parametric` into the master? We are working on something that will be nice to enforce parametricity.

**Comment by Saizan** [comment]
No, I don't believe there is a plan to merge this branch.

However @antoinevanmuylder is working on an implementation of of a different parametric theory.

**Comment by andreasabel** [comment]
Feels like this ship has sailed.
